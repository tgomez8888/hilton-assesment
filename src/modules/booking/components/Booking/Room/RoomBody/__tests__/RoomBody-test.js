import React from "react";
import { mount } from "enzyme";
import {RoomBody} from "..";

describe("RoomBody", () => {
  function render({ room, onChange }) {
    return mount(<RoomBody room={room} onRoomValuesChange={onChange}/>);
  }

  test("renders disabled fields when room is not selected", () => {
    const roomNotSelected = { id: 1, selected: false, adults: 1, children: 0 };
    const wrapper = render({ room: roomNotSelected, onChange: jest.fn() });

    expect(wrapper.html()).toMatchSnapshot();
  });

  test("renders enabled fields when room is selected", () => {
    const roomNotSelected = { id: 1, selected: true, adults: 1, children: 0 };
    const wrapper = render({ room: roomNotSelected, onChange: jest.fn() });

    expect(wrapper.html()).toMatchSnapshot();
  });

  test("onChange is called when adults dropdown value changes", () => {
    const testFunction = jest.fn();
    const onChange = e => testFunction(e.target.name, e.target.value);

    const wrapper = render({
      room: { id: 5, selected: true, adults: 2, children: 1 },
      onChange
    });
    wrapper
      .find("select[name='adults']")
      .first()
      .simulate("change", { target: { name: "adults", value: "1" } });

    expect(testFunction).toBeCalledWith("adults", "1");    
  });

  test("onChange is called when chilren dropdown value changes", () => {
    const testFunction = jest.fn();
    const onChange = e => testFunction(e.target.name, e.target.value);

    const wrapper = render({
      room: { id: 5, selected: true, adults: 2, children: 1 },
      onChange
    });
    wrapper
      .find("select[name='children']")
      .first()
      .simulate("change", { target: { name: "children", value: "1" } });

    expect(testFunction).toBeCalledWith("children", "1");
  });
});
