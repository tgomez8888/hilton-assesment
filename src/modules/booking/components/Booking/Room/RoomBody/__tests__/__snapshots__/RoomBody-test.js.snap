// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomBody renders disabled fields when room is not selected 1`] = `"<div class=\\"card-body\\"><div class=\\"column guess-selector\\"><label>Adults (18+)</label><select name=\\"adults\\" disabled=\\"\\"><option value=\\"1\\">1</option><option value=\\"2\\">2</option></select></div><div class=\\"column guess-selector\\"><label>Children (0-17)</label><select name=\\"children\\" disabled=\\"\\"><option value=\\"0\\">0</option><option value=\\"1\\">1</option><option value=\\"2\\">2</option></select></div></div>"`;

exports[`RoomBody renders enabled fields when room is selected 1`] = `"<div class=\\"card-body\\"><div class=\\"column guess-selector\\"><label>Adults (18+)</label><select name=\\"adults\\"><option value=\\"1\\">1</option><option value=\\"2\\">2</option></select></div><div class=\\"column guess-selector\\"><label>Children (0-17)</label><select name=\\"children\\"><option value=\\"0\\">0</option><option value=\\"1\\">1</option><option value=\\"2\\">2</option></select></div></div>"`;
